Library/Homebrew/test/test_integration_cmds.rb

```diff
@@ -42,8 +42,9 @@ def cmd_output(*args)
     cmd_args << "-rintegration_mocks"
     cmd_args << (HOMEBREW_LIBRARY_PATH/"../brew.rb").resolved_path.to_s
     cmd_args += args
+    cmd_exec = [RUBY_PATH] + cmd_args
     Bundler.with_original_env do
-      ENV["HOMEBREW_BREW_FILE"] = HOMEBREW_PREFIX/"bin/brew"
+      ENV["HOMEBREW_BREW_FILE"] = cmd_exec.join " "
```

Comment:
Assuming this is what `cmd_id_from_args`/`HOMEBREW_INTEGRATION_TEST` are doing: the arguments here will be handled correctly.